### PR TITLE
[prometheus-statsd-exporter] add first release

### PR DESCRIPTION
**Pull request by scDisorder**
Signed-off-by: disorder <disordervlz@gmail.com>

#### What this PR does / why we need it:
- I needed one, so I made it. Just though to share it will be nice.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by scDisorder** [comment]
hmm, probably the issue that helm actually does not runs without mappings specified...
wait a minute

**Comment by torstenwalter** [comment]
Can you add yourself to CODEOWNERS file?

File: charts/prometheus-statsd-exporter/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+```console
+# Helm 3
+$ helm show values prometheus-community/prometheus-statsd-exporter
+```
+
+### Statsd Exporter Server
+
+- Use early created ConfigMap with file `statsd.mappingConf` contained in data or specify mapping values in `statsd.mappingConfig`
+
+### Helm 2
```
Comment (torstenwalter) [inline charts/prometheus-statsd-exporter/README.md:64]: Not sure if we still need to mention it as it's now officially deprecated.

**Review by torstenwalter — APPROVED** [review]

File: charts/prometheus-statsd-exporter/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# prometheus-statsd-exporter
+
+Prometheus Exporter for [Statsd](https://github.com/statsd/statsd) metrics.
+
+This chart bootstraps a [Statsd Exporter](https://github.com/prometheus/statsd_exporter) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.16+ with Beta APIs enabled
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/README.md:9]:
```suggestion
- Kubernetes 1.16+ with Beta APIs enabled
- Helm 3+
```

File: charts/prometheus-statsd-exporter/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+```
+
+_See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation._
+
+## Install Chart
+
+```console
+# Helm 3
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/README.md:23]: No need for this comment anymore 🙂
```suggestion
```

File: charts/prometheus-statsd-exporter/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+_See [configuration](#configuration) below._
+
+_See [helm install](https://helm.sh/docs/helm/helm_install/) for command documentation._
+
+## Uninstall Chart
+
+```console
+# Helm 3
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/README.md:34]:
```suggestion
```

File: charts/prometheus-statsd-exporter/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+This removes all the Kubernetes components associated with the chart and deletes the release.
+
+_See [helm uninstall](https://helm.sh/docs/helm/helm_uninstall/) for command documentation._
+
+## Upgrading Chart
+
+```console
+# Helm 3
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/README.md:45]:
```suggestion
```

File: charts/prometheus-statsd-exporter/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+_See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
+
+## Configuring
+
+See [Customizing the Chart Before Installing](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing). To see all configurable options with detailed comments, visit the chart's [values.yaml](./values.yaml), or run these configuration commands:
+
+```console
+# Helm 3
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/README.md:56]:
```suggestion
```

File: charts/prometheus-statsd-exporter/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v2
+name: prometheus-statsd-exporter
+description: A Helm chart for prometheus stats-exporter
+version: 0.1.0
+appVersion: 0.18.0
+home: https://github.com/prometheus/statsd_exporter
+sources:
+  - https://github.com/prometheus/statsd_exporter
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/Chart.yaml:8]: For Artifact Hub. This link is already used for `home`, let's just add the chart source so people know where to find it. If there are other relevant source links we can add them here too, and label them accordingly.
```suggestion
  - https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-statsd-exporter
annotations:
  "artifacthub.io/links": |
    - name: Chart Source
      url: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-statsd-exporter
```

File: charts/prometheus-statsd-exporter/values.yaml [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+# deploymentRevisionHistoryLimit: 10
+
+image:
+  repository: prom/statsd-exporter
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "latest"
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/values.yaml:12]: Default tag should be pinned.
```suggestion
  tag: v0.18.0
```

**Review by scottrigby — CHANGES_REQUESTED** [review]

**Comment by scDisorder** [comment]
kkay, I should wait probably

**Review by scottrigby — APPROVED** [review]

File: charts/prometheus-statsd-exporter/templates/ingress.yaml [hunk 9]
```diff
@@ -0,0 +1,41 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "prometheus-statsd-exporter.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/templates/ingress.yaml:8]: One more thing, we don't need this check if a prerequisite of this chart is
> Kubernetes 1.16+ with Beta APIs enabled

```suggestion
apiVersion: networking.k8s.io/v1beta1
```

**Review by scottrigby — CHANGES_REQUESTED** [review]

**Comment by scDisorder** [comment]
should we add `networking.k8s.io/v1` instead of `v1beta` for Kubernetes 19+ ?

**Comment by scDisorder** [comment]
> should we add `networking.k8s.io/v1` instead of `v1beta` for Kubernetes 19+ ?

well, I think this is more about helm backwards compatibility actually

**Comment by scDisorder** [comment]
woah, missclick

**Review by scottrigby — DISMISSED** [review]
This lgtm. If you want to do a semver check for 1.19+ and switch to that version you are welcome to do that. This version is still supported though in 1.19 so up to you.

**Review by monotek — APPROVED** [review]
@scottrigby 
any reason this was not merged?

**Comment by scottrigby** [comment]
@monotek I was just being patient waiting to see if they wanted to make a change based on my reply to their semver question.

**Comment by monotek** [comment]
ah, ok. i was just wondering, because it was approved and not merged. so i wont merge it too :D

**Comment by scDisorder** [comment]
not sure if default pathType must be `Prefix`, it is common thing to use `ImplementationSpecific` but i see no reasons up here

**Comment by scDisorder** [comment]
1037d952b8ecfe75

File: charts/prometheus-statsd-exporter/templates/ingress.yaml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  {{- end }}
+  rules:
+    {{- range .Values.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ . }}
+            {{- if semverCompare ">=1.19.0-0" $kubeTargetVersion }}
+            pathType: {{ .Values.ingress.pathType | default "Prefix" }}
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/templates/ingress.yaml:38]: @scDisorder I'm not really familiar with `pathType` yet apart from reading the docs, but there it says `ImplementationSpecific` is the default. Should this default follow, or is there a reason we want `Prefix` here?

File: charts/prometheus-statsd-exporter/values.yaml [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+  annotations: {}
+
+ingress:
+  enabled: false
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  hosts:
+    - host: chart-example.local
+      #  pathType: Prefix # only Kubernetes v1.19+
```
Comment (scottrigby) [inline charts/prometheus-statsd-exporter/values.yaml:89]: See pathType default comment above

File: charts/prometheus-statsd-exporter/templates/ingress.yaml (hunk 10 above)
Comment (scDisorder) [reply charts/prometheus-statsd-exporter/templates/ingress.yaml]: wow, probably i misudnerstood the doc or took it at the wrong time
fixed

**Review by scottrigby — APPROVED** [review]
lgtm 👏 nice work

**Pull request merged**